### ISSUE TITLE
Offline building with v3.0

### ISSUE THREAD

**Issue by nodje** [ansible/ansible-builder#606]
I'm not able to update my configuration to build with v3.0 since I'm offline.

There was an option to use a local `ubi.repo` file that apparently disappeared in v3.0. Or at least I can't find it.

I use 
```
options:
  package_manager_path: /usr/bin/microdnf
```
since I use Red Hat images, but the build is stuck looking for some public repo that is not accessible for me:
```
Downloading metadata...
error: cannot update repo 'ubi-9-baseos-rpms': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried; Last error: Curl error (6): Couldn't resolve host name for https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/x86_64/baseos/os/repodata/repomd.xml [Could not resolve host: cdn-ubi.redhat.com]
```

How can I provide a local repo option that would lead to something 1.0 equivalent in Containfile:
```
# this section added
ADD ubi.repo /etc/yum.repos.d/ubi.repo
ADD pip.conf /etc/pip.conf
# end additions
```

**Comment by Akasurde** [comment]
Please take a look at https://ansible.readthedocs.io/projects/builder/en/latest/scenario_guides/scenario_copy/#copying-arbitratory-files-to-ee for copying arbitrary files.

tl;dr - 

```
additional_build_files:
  # copy arbitrary files next to this EE def into the build context - we can refer to them later...
  - src: files/ubi.repo
    dest: configs
  - src: files/pip.conf
    dest: configs

additional_build_steps:
  prepend_base:
    - COPY _build/configs/ubi.repo /etc/yum.repos.d/ubi.repo
    - COPY _build/configs/pip.conf /etc/pip.conf
```

**Comment by nodje** [comment]
easy baby, thanks for this!

**Comment by nodje** [comment]
For the record, if ones want to provide local rpm repos, it is needed to delete the existing ones:

```
additional_build_steps:
  prepend_base:`
    - RUN rm -rf /etc/yum/repos.d/*
    - COPY _build/configs/ubi.repo /etc/yum.repos.d/ubi.repo
    - COPY _build/configs/pip.conf /etc/pip.conf
```

**Comment by nodje** [comment]
somehow related to the Offline usage topic:

I'm using a local Automation Hub as repository and registry.
It's HTTPS certificate is sign by a local CA.

I have to pass the rootCA.crt as was done with Red Hat provided for ansible-builder v1.
ContainerFile v1
```
# this section added
ADD ansible.cfg /etc/ansible/ansible.cfg
ADD hub-root.crt /etc/pki/ca-trust/source/anchors/hub-root.crt
RUN update-ca-trust
# end additions
```
this above worked fine.


Now I'm using:
```
additional_build_steps:
  prepend_base:
    - COPY _build/configs/rootCA.crt /etc/pki/ca-trust/source/anchors/rootCA.crt
    - RUN update-ca-trust
```
which generates, as expected
```
# Base build stage
FROM $EE_BASE_IMAGE as base
USER root
ARG EE_BASE_IMAGE
ARG PYCMD
ARG PYPKG
ARG PKGMGR_PRESERVE_CACHE
ARG ANSIBLE_GALAXY_CLI_COLLECTION_OPTS
ARG ANSIBLE_GALAXY_CLI_ROLE_OPTS
ARG ANSIBLE_INSTALL_REFS
ARG PKGMGR

ENV TZ="Europe/Paris"
RUN rm -rf /etc/yum.repos.d/*
COPY _build/configs/ubi9.repo /etc/yum.repos.d/ubi9.repo
COPY _build/configs/pip.conf /etc/pip.conf
COPY _build/configs/rootCA.crt /etc/pki/ca-trust/source/anchors/rootCA.crt
RUN update-ca-trust
RUN $PKGMGR install $PYPKG -y ; if [ -z $PKGMGR_PRESERVE_CACHE ]; then $PKGMGR clean all; fi
RUN $PYCMD -m ensurepip
RUN $PYCMD -m pip install --no-cache-dir $ANSIBLE_INSTALL_REFS
COPY _build/scripts/ /output/scripts/
COPY _build/scripts/entrypoint /opt/builder/bin/entrypoint
```

And it looks like it should work but somehow I keep getting an SSL  certificate validation error:
```
[2/4] STEP 16/16: RUN ANSIBLE_GALAXY_DISABLE_GPG_VERIFY=1 ansible-galaxy collection install $ANSIBLE_GALAXY_CLI_COLLECTION_OPTS -r requirements.yml --collections-path "/usr/share/ansible/collections"
[WARNING]: Skipping Galaxy server
https://s01vl9989234.fr.net.intra/api/galaxy/content/published/. Got an
unexpected error when getting available versions of collection
ansible.controller: Unknown error when attempting to call Galaxy at
'https://s01vl9989234.fr.net.intra/api/galaxy/content/published/api': <urlopen
error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get
local issuer certificate (_ssl.c:1129)>. <urlopen error [SSL:
CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local
issuer certificate (_ssl.c:1129)>
...
```

any idea on what could be occuring here?

**Comment by nodje** [comment]
I've read the man page for `update-ca-trust` and checked the resulting `/etc/pki/tls/certs/ca-bundle.crt file.
My certificate is in it at the top of the list.

Everything seems fine, but either curl or ansible-galaxy all fails with the same SSL error

**Comment by budgester** [comment]
I just remove the base image repo which I believe then means it falls back to using the hosts repo's

```
additional_build_steps:
  prepend_base:
    - RUN rm -f /etc/yum.repos.d/ubi.repo
```
